src/async_impl/request.rs

```diff
@@ -346,8 +347,8 @@ impl RequestBuilder {
     /// ```
     pub fn send(self) -> impl Future<Item = Response, Error = ::Error> {
         match self.request {
-            Ok(req) => self.client.execute(req),
-            Err(err) => Pending::new_err(err),
+            Ok(req) => Either::A(self.client.execute(req)),
+            Err(err) => Either::B(Pending::new_err(err)),
```

Comment:
Hm! I'm guessing this is because the `impl Future` from `execute` means the type is erased, and so the return types no longer match?

It'd be better to not include the extra `Either` branch. Perhaps this could call `execute_request` instead, if that were made `pub(super)`?